File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        pascal = []
        for i in range(numRows):
            line = [1]*(i+1)
            for j in range(1,i):
                line[j]=pascal[i-1][j-1] + pascal[i-1][j]
            pascal.append(line)
        return pascal
        

File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>pascal;
        for(int i=0;i<numRows;i++){
            vector<int>line(i+1,1);
            for(int j=1;j<i;j++){
                line[j] = pascal[i-1][j-1] + pascal[i-1][j];
            }
            pascal.push_back(line);
        }
        return pascal;
    }
};